docs/gke-autopilot.md

```diff
@@ -0,0 +1,268 @@
+demo].
+
+> Whether you use Cloud Console or the command line to create the cluster, you
+will still need to use `kubectl` at the command line to deploy the application.
+>
+> The instructions assume you already have [Google Cloud SDK (gcloud)] and
+[kubectl] (which can be optionally installed as a `gcloud` component).
+>
+> If you prefer, you can use [Cloud Shell] for a hosted terminal environment
+that already has these tools installed for you.
+
+
+### Select or create a project using Cloud Console
+
+1. In the Google Cloud Console, go to the [project selector] page. Either select
+an existing project or create a new one.
+
+1. Next, using the left navigation menu, choose **Kubernetes Engine**.
+
+1. If prompted, click to **ENABLE** the Kubernetes API.
+
+1. On the **Kubernetes Engine Clusters** page, click the **CREATE** button.
+
+1. The **Create Cluster** dialog will prompt you to select a cluster mode. Click
+the **CONFIGURE** button.
+
+1. You will be prompted to **Create an Autopilot cluster**. You can accept all
+the defaults.
+
+1. Once the cluster is created, click on the name link to go to the cluster
+**DETAILS** tab. Click the **CONNECT** button to get the command you'll use to
+connect to the cluster using the command line. Copy the command and paste it
+to your command line and press enter.
+
+Now you're ready to deploy the demo.
+
+### Select or create a project using the command line
+
+The following steps are purely administrative. You need to log in and identify
+the project you want to work with.
+
+```text
+gcloud auth login
+```
+
+Then either choose an existing project or create a new one.
+
+#### Select an existing project
+
+```text
+PROJECT_ID=my-project-id
+gcloud services enable container.googleapis.com --project $PROJECT_ID
+```
+
+This assumes you have billing set for the project. If not, see the next section
+and look for the `gcloud beta billing` command to link billing up.
+
+#### Create a new project
+
+> You will need to know your [billing ID].
+
+```text
+PROJECT_NAME=my-project-name
+gcloud projects create $PROJECT_NAME
+```
+
+The output will print the assigned project ID associated with the project name.
+Use the following command to get the project ID and store it to `PROJECT_ID`:
+
+```text
+export PROJECT_ID=$(gcloud projects list --filter NAME=$PROJECT_NAME --format='value(PROJECT_ID)')
+```
+
+#### Set billing for the project.
+
+```text
+gcloud beta billing projects link tonypujals-autopilot-demo --billing-account=XXXXXX-XXXXXX-XXXXXX
+```
+
+#### Enable the containers API for your project
+
+```text
+gcloud services enable container.googleapis.com --project $PROJECT_ID
+```
+
+#### Create the Autopilot cluster
+
+By creating an Autopilot cluster (with `create-auto`) you only need to give the
+cluster a name. You can't specify any options that configure cluster nodes.
+
+```text
+gcloud container clusters create-auto microservices-demo
```

Comment:
When I ran:
```
gcloud container clusters create-auto microservices-demo
```
I got the error:
```
ERROR: (gcloud.container.clusters.create-auto) ResponseError: code=400, message=Autopilot clusters must be regional clusters.
```
I think problem was caused by:
* having set a default `[compute/zone]` value
* and the fact that autopilot is [regional, not zonal](https://cloud.google.com/kubernetes-engine/docs/concepts/types-of-clusters#config-choices).

I was able to get it to work by either:
* specifying the region: `gcloud container clusters create-auto microservices-demo --region=us-east1`
* or unsetting the default `[compute/zone]` value: `gcloud config unset compute/zone`


So maybe let's add something similar to `--region=us-central1` to the command here.